test/src/main/java/org/apache/accumulo/test/CleanWalIT.java

```diff
@@ -136,11 +136,13 @@ private int countLogs(String tableName, Connector conn) throws TableNotFoundExce
       log.debug("Saw {}={}", entry.getKey(), entry.getValue());
       count++;
     }
+    scanner.close();
     return count;
   }
 
   int count(String tableName, Connector conn) throws Exception {
     Scanner s = conn.createScanner(tableName, Authorizations.EMPTY);
+    s.close();
```

Comment:
This shouldn't be closed before it is used.

Reply:
Right so in these cases would a try block work instead? since it should close on its on in that case? or do I just close it in the method it's thrown to?

Reply:
This is a good example to highlight three different possibilities. In all three cases, the scanner instance never leaves the method (we never do `return s;` or `callOtherMethodToFinish(s);`), so we should always try to close it within the method.

The first is the best because it guarantees the scanner is closed before the method returns, but also doesn't require all the complicated syntax of the old finally block in the second example. The third example is the worst, because it can leave the scanner unclosed.

```java
    // using try-with-resources
    try (Scanner s = conn.createScanner(tableName, Authorizations.EMPTY)) {
        return Iterators.size(s.iterator());
    }
```

```java
    // make sure it's closed, using old finally syntax
    Scanner s = null;
    try {
      s = conn.createScanner(tableName, Authorizations.EMPTY);
      return Iterators.size(s.iterator());
    } finally {
      if (s != null) {
        s.close;
      }
    }
```

```java
      // does not guarantee it is closed if it has an exception, but does try before it returns from the method
      Scanner s = conn.createScanner(tableName, Authorizations.EMPTY);
      int size = Iterators.size(s.iterator());
      s.close();
      return size;
```

Reply:
So I will go through and make the the scanners like the 1st example, 2nd if necessary and I don't think the 3rd way is ever necessary.